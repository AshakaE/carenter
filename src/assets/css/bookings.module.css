.container {
  @apply m-8 bg-green-200 border-green-200 border bg-opacity-10;
}

.gridHead {
  @apply grid grid-cols-7 items-center font-bold border-green-200 border-b py-3 px-3 bg-tLight text-white;
}

.gridBox {
  @apply grid grid-cols-7 items-center font-medium px-3 border-green-200 border-b py-2;
}

.action {
  @apply col-span-2;
}

.btn {
  @apply py-1 px-1 my-2 mx-1 w-auto text-white text-sm text-center font-normal rounded-lg focus:outline-none cursor-pointer uppercase bg-pLight hover:bg-pDark font-medium duration-100;
}

.btnDel {
  @apply py-1 px-1 w-auto text-white text-sm text-center font-normal rounded-lg focus:outline-none cursor-pointer uppercase bg-red-600 hover:bg-red-800 font-medium duration-100;
}

.form {
  @apply px-80;
}

.cardText {
  @apply ml-[41%] mt-14 font-head text-gray-700 text-6xl m-auto;
}

.links {
  @apply flex w-4/12 ml-[31%] justify-around mt-5 font-head text-sLight text-xl;
}

.link {
  @apply hover:text-sDark underline;
}
