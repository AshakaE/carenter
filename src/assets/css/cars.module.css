.container {
  @apply flex flex-col bg-white m-auto;
}

.header {
  @apply flex py-5 lg:px-20 md:px-10 px-5 lg:mx-40 md:mx-20 mx-5 font-bold text-5xl text-gray-800 mt-10 mb-5 font-head;
}

.card {
  @apply flex overflow-x-scroll pb-10 overscroll-none;
}

.cardInner {
  @apply flex flex-nowrap lg:ml-40 md:ml-20 ml-10 mb-24;
}

.cardI {
  @apply inline-block px-3;
}

.cardBox {
  @apply w-[450px] overflow-hidden rounded-lg shadow-md bg-white hover:shadow-xl transition-shadow duration-300 ease-in-out;
}

.cardText {
  @apply flex px-2 font-head text-gray-700;
}

.btn {
  @apply py-1 px-4 w-full mt-1 text-white text-center font-normal text-xl rounded-lg focus:outline-none cursor-pointer uppercase bg-pLight hover:bg-pDark font-medium duration-100;
}

.links {
  @apply flex w-4/12 ml-[4%] justify-around pb-20 font-head text-sLight text-xl;
}

.link {
  @apply hover:text-sDark underline;
}
