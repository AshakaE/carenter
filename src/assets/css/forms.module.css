.inputDiv {
  @apply flex justify-between py-1;
}

.inputBox {
  @apply w-full py-1 text-center border-[0.1px] border-pLight focus:outline-none rounded-md;
}

.info {
  @apply text-center italic;
}
