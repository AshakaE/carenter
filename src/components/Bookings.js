import React from 'react';

const Bookings = () => {
  const one = 1; //eslint-disable-line
  return (
    <>
      <div>Bookings</div>
    </>
  );
};

export default Bookings;
