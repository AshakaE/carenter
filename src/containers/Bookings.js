import React, { useEffect } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { getBookings } from '../redux/actions';
import Loading from '../components/Loading';
import BookingsCard from '../components/BookingsCard';
import tables from '../assets/css/bookings.module.css';

const Bookings = (props) => {
  const { getBookings, bookings, loading } = props;

  useEffect(() => {
    getBookings();
  }, []);

  if (loading) {
    return (
      <Loading />
    );
  }

  return (
    <>
      <h1 className={tables.cardText}>Bookings</h1>
      <div className={tables.links}>
        <div className={tables.link}>
          <Link to="/cars">Cars gallery</Link>
        </div>
        <div className={tables.link}>
          <Link to="/user">My Account</Link>
        </div>
      </div>
      <div className={tables.container}>
        <div className={tables.gridHead}>
          <div>Title</div>
          <div>Car make</div>
          <div>Car model</div>
          <div>Created by</div>
          <div>Price ($)</div>
          <div>Date</div>
          <div>Actions</div>
        </div>
        {Object.values(bookings).map((item) => (
          <BookingsCard key={item.id} item={item} />
        ))}
      </div>
    </>
  );
};

const mapStateToProps = ({
  bookingState: { bookings, loading },
}) => ({ loading, bookings });

Bookings.propTypes = {
  loading: PropTypes.bool.isRequired,
  bookings: PropTypes.arrayOf(PropTypes.object),
  getBookings: PropTypes.func.isRequired,
};

Bookings.defaultProps = {
  bookings: [],
};

export default connect(mapStateToProps, { getBookings })(Bookings);
