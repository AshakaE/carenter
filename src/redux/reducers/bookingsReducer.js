import { SET_LOADING, GET_BOOKINGS } from '../actions';

const defaultState = {
  loading: false,
  bookings: [],
};

const bookingsReducer = (state = defaultState, action) => {
  if (action.type === SET_LOADING) {
    return { ...state, loading: true };
  }
  if (action.type === GET_BOOKINGS) {
    return { ...state, loading: false, bookings: action.payload };
  }
  return state;
};

export default bookingsReducer;
