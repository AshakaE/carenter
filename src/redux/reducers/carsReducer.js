import { SET_LOADING, GET_CARS } from '../actions';

const defaultState = {
  loading: false,
  cars: [],
};

const carsReducer = (state = defaultState, action) => {
  if (action.type === SET_LOADING) {
    return { ...state, loading: true };
  }
  if (action.type === GET_CARS) {
    return { ...state, loading: false, cars: action.payload };
  }
  return state;
};

export default carsReducer;
