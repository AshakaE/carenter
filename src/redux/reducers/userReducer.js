import { SET_LOADING, GET_USER } from '../actions';

const defaultState = {
  loading: false,
  user: '',
};

const userReducer = (state = defaultState, action) => {
  if (action.type === SET_LOADING) {
    return { ...state, loading: true };
  }
  if (action.type === GET_USER) {
    return { ...state, loading: false, user: action.payload };
  }
  return state;
};

export default userReducer;
