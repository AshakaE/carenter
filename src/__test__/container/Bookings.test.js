import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { createStore, combineReducers, applyMiddleware } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import thunk from 'redux-thunk';
import '../../index.css';
import carsReducer from '../../redux/reducers/carsReducer';
import bookingsReducer from '../../redux/reducers/bookingsReducer';
import userReducer from '../../redux/reducers/userReducer';
import Bookings from '../../containers/Bookings';

const middleware = [thunk];
const store = createStore(
  combineReducers({
    carState: carsReducer,
    userState: userReducer,
    bookingState: bookingsReducer,
  }),
  composeWithDevTools(applyMiddleware(...middleware)),
);

describe('Bookings', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Bookings store={store} />
        </BrowserRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
