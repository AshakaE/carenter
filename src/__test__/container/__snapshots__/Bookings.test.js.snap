// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bookings renders correctly 1`] = `
Array [
  <h1
    className="cardText"
  >
    Bookings
  </h1>,
  <div
    className="links"
  >
    <div
      className="link"
    >
      <a
        href="/cars"
        onClick={[Function]}
      >
        Cars gallery
      </a>
    </div>
    <div
      className="link"
    >
      <a
        href="/user"
        onClick={[Function]}
      >
        My Account
      </a>
    </div>
  </div>,
  <div
    className="container"
  >
    <div
      className="gridHead"
    >
      <div>
        Title
      </div>
      <div>
        Car make
      </div>
      <div>
        Car model
      </div>
      <div>
        Created by
      </div>
      <div>
        Price ($)
      </div>
      <div>
        Date
      </div>
      <div>
        Actions
      </div>
    </div>
  </div>,
]
`;
